EIPS/eip-4950.md

```diff
@@ -0,0 +1,78 @@
+
+The entangled tokens have important applications such us keeping permanent contact of two users.
+
+An use case is to keep contact between an artist and an buyer of its NFTs. If an artist T has created a digital piece of art P with an NFT with wallet, then T creates 2 entangled tokens A and B so that he keeps A and transfer B to the wallet of P. By construction of entangled tokens, only one transfer is possible for them, thus the artist proofs he’s been the creator of P by sending a transaction to A that is visible from B. Otherwise, the owner of P might check the authenticity of the artist by sending a transaction to B so that the artist might proof by showing the outcome out of A.
+
+An extension of this first use case is when one user U and its creation in the form of an NFT T with wallet then the user U creates and transfers an entangled token A to T, and U keeps the entangled token B in the U’s wallet.  
+
+In both cases it is advisable that the entangled tokens cannot be re-transferred, that means that only once can be transferred from their supply to the definite wallet, of the NFT or of the user.
+
+These applications of entangled tokens are envisaged to be useful for 
+1.	NFT authorship / art creation
+2.	Distribution of royalties by the creator.
+3.	Authenticity of a work of art: creation limited to the author (e.g. only 1000 copies if there are 1000 1000 entangled tokens in that NFT).
+4.	Usowners
+5.	Reformulation of property rights: the one who owns the property receives it without having to follow in the footsteps of the owners.
+6.	Identity: Only those credentials that have an entangled token with you are related to you.
+7.	Vreservers (value-reservers).
+
+## Backwards Compatibility
+No backwards compatibility issues devised.
+
+## Reference Implementation
+Mint two tokens, and only two, at the contract constructor, and set the `minted` property to true:
+
+```solidity
+    bool private _minted;
+
+    constructor(string memory name, string memory symbol, string memory base_uri) ERC721(name, symbol) {
+        baseUri = base_uri;
+        mint(msg.sender,0);
+        mint(msg.sender,1);
+        _minted = true;
+    }
+```
+
+Add additional functions to allow both NFT user owners to operate with other ERC20 tokens owned by the contract:
+
+```solidity
+    function spendTokens(WalletNFT2 wallet, IERC20 token_address, address recipient, uint256 amount) public onlyOwners {
+        wallet.transferTokens(token_address, recipient, amount);
+    }
+```

```

Comment:
I don't think this reference implementation gives enough information about the standard itself. What is a `WalletNFT2`?

Reply:
Corrected